### PR TITLE
ValidClassName sniffs: Add trait support.

### LINKED ISSUE #1152: Add `T_TRAIT` to `Squiz.Classes.ValidClassNameSniff`.
Now this snuff checks only interfaces and classes, be good have similar sniff behaviour to trait.

### PR DESCRIPTION
**Pull request by jrfnl**
Includes unit tests.

Fixes #1152

### CODE REVIEW

File: CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidClassNameSniff.php [hunk 1]
```diff
@@ -42,6 +42,7 @@ public function register()
         return array(
                 T_CLASS,
                 T_INTERFACE,
+                T_TRAIT,
```
Comment (aik099) [inline CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidClassNameSniff.php]: Not sure if PEAR tells anything about traits in it.

Comment (jrfnl) [inline CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidClassNameSniff.php]: Nor me to be fair. What about if I make it a toggleable option? I.e. add a public property `checkTraits` which defaults to `false` (old behaviour), but can be changed via a custom ruleset ?

Comment (gsherwood) [inline CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidClassNameSniff.php]: The PEAR standard was written before many language features, like traits, so I keep it updated from time to time in this same way. I don't see any problem with this change.

**Comment by gsherwood** [comment]
Thanks for the PR

**Comment by jrfnl** [comment]
You're welcome ;-)

**Pull request merged**